### PR TITLE
Replaces Mules with a Cargo Hauler and Rollerskates on Meta

### PR DESCRIPTION
**Pull request by Son-of-Space**
## About The Pull Request
This PR replaces the two mulebots in the warehouse on Meta with a cargo variant of the ripley and two pairs of rollerskates. The cargo variant of the Ripley has only two maximum equipment slots, starts with 50 integrity, and a smaller power cell. A recharging station for the hauler is available in cargo. (Apparently the station doesn't need a computer to function).

I've left the infrastructure for the mules in case someone somehow gets one or in case someone wants to expand on this and make mules something buildable or purchasable.

EDIT: I have been advised to leave at least one Mulebot on the map, and thus will.

EDIT: I have also reduced the maximum integrity of the mech to 100 (Which I thought I did by editing its integrity value, but didn't) - it should further reduce people repurposing it as a weapon.

## Why It's Good For The Game

Mulebots have historically been incredibly poor performing. They're slow, unresponsive, clunky, and when they do work properly, they only work in batches of one crate and require you to radio in that something has been delivered. More often than not, since you aren't dealing with people, your deliveries are left forgotten under plastic flaps in a department. Most of the time, people either ignore the Mulebots and leave them to gather dust in the corner or hack them to use them as murder machines instead of for their primary purpose.

I've been hearing that people feel cargo techs don't have has much to do during the shift as people would like, so I thought it would be a good idea to give them some tools that will help them do their job in a way that interacts with the crew. The rollerskates are a great way to get around if you're hauling singular crates (and are kind of funny if you think of the techies as busboys). The cargo hauler should make it much easier to make batch deliveries personally and I've attempted to minimize any abuses people may have with it by limiting its equipment slots.

If the changes are well received, I think it would be cool to eventually replace the cargo hauler with a forklift vehicle.

## Changelog
:cl: Son of Space, Fikou
add: A RIPLEY variant with its own set of sprites.
balance: Mules on Meta have been replaced by a cargo hauler and a few pairs of rollerskates
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
👎 
the proper solution here is to make mulebots less garbage, either by making them remote-controlled, giving them onboard cameras, or some other less-discussed solution.

**Comment by Son-of-Space** [comment]
> 
> 
> 👎
> the proper solution here is to make mulebots less garbage, either by making them remote-controlled, giving them onboard cameras, or some other less-discussed solution.

I heavily disagree. Mulebots actively discourage crew interaction in an uninteresting way and I don't see a way to fix them that anybody would want to do - especially since they've been the exact same for years.

**Comment by Son-of-Space** [comment]
Crouching tiger, Hidden Mulebot

image

As per mothblocks' recommendation to keep one on map.

**Comment by Zytolg** [comment]
Now it's your turn to "borrow my ideas" 😄 


> Crouching tiger, Hidden Mulebot
> 
> image
> 
> As per mothblocks' recommendation to keep one on map.

I like that disused mulebots have a Asian style name. Next one should be called _Leaping Rabbit_

**Comment by Son-of-Space** [comment]
> 
> 
> Now it's your turn to "borrow my ideas" 😄
> 
> > Crouching tiger, Hidden Mulebot
> > image
> > As per mothblocks' recommendation to keep one on map.
> 
> I like that disused mulebots have a Asian style name. Next one should be called _Leaping Rabbit_

I will unironically go back into the editor right now and change the mulebot's name to Leaping Rabbit

**Comment by MrDoomBringer** [comment]
hmm okay the maint mulebot has convinced me
maybe later mulebots can make a comeback when they're less shit

**Comment by iwishforducks** [comment]
Mulebots are (were) taking our jobs away!

Mulebots essentially took away job content if you decided to use them. Removing them and giving Cargo better tools to deliver crates is a HUGE step in the right direction.

**Comment by Jared-Fogle** [comment]
i only asked for it because pai emagged mulebots are hillarious btw

though might want to check that the sentience event doesn't just trap them in that room :)

**Comment by iwishforducks** [comment]
replace the grille with a plastic flap and windoor and i think that'd would fix the sentience trap problem

**Comment by ATH1909** [comment]
> The cargo variant of the Ripley has only two maximum equipment slots, starts with 50% integrity, and a smaller power cell.

Oh, c'mon now, give them the normal number of maximum equipment slots, there's no real reason to nerf that.

**Comment by itseasytosee** [comment]
You could give it two hydraulic clamps like in the movies.

**Comment by MrDoomBringer** [comment]
@ATH1909 the 'real reason' is cargo does not need to make robotics even more useless with a free mech that miners would probably end up using

File: code/modules/vehicles/mecha/working/ripley.dm
Comment (Fikou) [inline code/modules/vehicles/mecha/working/ripley.dm:142]:
```suggestion
	desc = "An ailing, old, repurposed cargo hauler."
```

File: code/modules/vehicles/mecha/working/ripley.dm
Comment (Fikou) [inline code/modules/vehicles/mecha/working/ripley.dm:143]: this name sucks ass just keep it default of think of something better

File: code/modules/vehicles/mecha/working/ripley.dm
Comment (GunHog) [reply code/modules/vehicles/mecha/working/ripley.dm:144]: Nothing wrong with someone wanting to stick an upgrade on it. It might be easier to ask Robotics for one item rather than a whole mech.

Comment (AlinaStarkova) [reply code/modules/vehicles/mecha/working/ripley.dm:144]: two hydrolic clamps. for two crate hauling

Comment (Son-of-Space) [reply code/modules/vehicles/mecha/working/ripley.dm:144]: Basically what GunHog said. They can make it a LITTLE better, but not much.

File: code/modules/vehicles/mecha/working/ripley.dm
Comment (GunHog) [reply code/modules/vehicles/mecha/working/ripley.dm:143]: Could just name it "Power Loader", while also removing "APLU" since that just means "Autonomous Power Loader Unit". "Hauler" works if you are feeling uncreative. Ultimately, the default name with something like "old", "early" or "prototype" prefixed would work too.

Comment (Son-of-Space) [reply code/modules/vehicles/mecha/working/ripley.dm:143]: better name coming right up

**Comment by ShibaInuLord** [comment]
> 
> 
> by the way, mules are already purchaseable in cargo, so you dont need to worry about that. it is not a concern even if we started with none, want pAI murder bots? just buy them, they have been buyable in cargo for years now, its just that nobody ever bought them

the more you know

**Comment by Tlaltecuhtli** [comment]
mulebots are fine as they are, really efficient when you give them a pai, i dont see why there is there a need to remove them from roundstart, hacking them and causing road kills is always  fun. i dont think anyone ever delivers  crate to departments outside to like botany bees creating a need of  dumb stuff like a crappy ripley that can walk 10 tiles before discharging and its slower than just walking on foot

**Comment by D4C-420** [comment]
Yea but those mule bots crack bones more than anything else. Want a fucked up corse? let a mule bot on em.

**Comment by ATH1909** [comment]
> @ATH1909 the 'real reason' is cargo does not need to make robotics even more useless with a free mech that miners would probably end up using

at least give it its own special name or something so it doesn't look identical to a normal ripley but have less stats for some reason

**Comment by cacogen** [comment]
I haven't read all this and while I dislike removing MULEbots it's true that encouraging hand deliveries and consequential things like tips is a good way of creating crew interaction and economy interaction

**Comment by Learner10** [comment]
MULEbots become really efficient when hacked. The main issue is that they just often leave their deliveries inside the flaps, when technically they're suppose to leave their deliveries on the tile with the windoor. If the windoor is left open, the Mulebot will receive and drop crates inside the department, and not in the flaps but the Mulebot doesn't seem to actually open it up.

The last time I used Mulebots efficiently, it was to deliver and retrieve canisters from Atmos to sell. It ends up being really efficient being able to send the bot to retrieve a package for you while you work on another.

**Comment by Son-of-Space** [comment]
> 
> 
> > @ATH1909 the 'real reason' is cargo does not need to make robotics even more useless with a free mech that miners would probably end up using
> 
> at least give it its own special name or something so it doesn't look identical to a normal ripley but have less stats for some reason

read the commits

File: code/modules/vehicles/mecha/working/ripley.dm
Comment (ATH1909) [inline code/modules/vehicles/mecha/working/ripley.dm:142]: you should probably directly mention the reduced equipment capacity in this description

Comment (Son-of-Space) [reply code/modules/vehicles/mecha/working/ripley.dm:142]: You're right.

**Comment by Son-of-Space** [comment]
ATH's review helped me realize that I only set the starting health of the ripley lower and not also its max health. I made it 100. I also added to the description for that ("its frame is rusted"). For the equipment, "Most of its equipment wires are frayed or missing"

**Comment by Son-of-Space** [comment]
Fikou will be giving me some sprites for Big Bess to properly distinguish it from others.

**Comment by Tlaltecuhtli** [comment]
does the mulebot even work if you removed his home beacon?

**Comment by AlinaStarkova** [comment]
> 
> 
> MULEbots become really efficient when hacked. The main issue is that they just often leave their deliveries inside the flaps, when technically they're suppose to leave their deliveries on the tile with the windoor. If the windoor is left open, the Mulebot will receive and drop crates inside the department, and not in the flaps but the Mulebot doesn't seem to actually open it up.
> 
> The last time I used Mulebots efficiently, it was to deliver and retrieve canisters from Atmos to sell. It ends up being really efficient being able to send the bot to retrieve a package for you while you work on another.

they travel to the beacon and then drop the crate in a direction, they dont open doors in its way no, thats why you need to add a windoor on the flap leading to maint and not leading to the dept. it *will* open that one to stand on the beacon

**Comment by Son-of-Space** [comment]
> 
> 
> does the mulebot even work if you removed his home beacon?

read the PR.

**Comment by Learner10** [comment]
> > MULEbots become really efficient when hacked. The main issue is that they just often leave their deliveries inside the flaps, when technically they're suppose to leave their deliveries on the tile with the windoor. If the windoor is left open, the Mulebot will receive and drop crates inside the department, and not in the flaps but the Mulebot doesn't seem to actually open it up.
> > The last time I used Mulebots efficiently, it was to deliver and retrieve canisters from Atmos to sell. It ends up being really efficient being able to send the bot to retrieve a package for you while you work on another.
> 
> they travel to the beacon and then drop the crate in a direction, they dont open doors in its way no, thats why you need to add a windoor on the flap leading to maint and not leading to the dept. it _will_ open that one to stand on the beacon

So it's a mapping issue then?

**Comment by TheBonded** [comment]
Restrict the useable equipments on it so it can only use the hydraulic clamp, armour and power generators. no reason to give cargo a roundstart way of drilling open armoury crates
Also, maybe because it has a rusted frame, you can see through some damaged parts and see what it has loaded on inspection? just a thought

I think this is a step in the right direction for cargo

**Comment by Son-of-Space** [comment]
> 
> 
> Restrict the useable equipments on it so it can only use the hydraulic clamp, armour and power generators. no reason to give cargo a roundstart way of drilling open armoury crates
> Also, maybe because it has a rusted frame, you can see through some damaged parts and see what it has loaded on inspection? just a thought
> 
> I think this is a step in the right direction for cargo

I'm nowhere near competent enough with byond code to implement what you're suggesting, but anybody else who comes around is welcome to make their own PR for it.

**Review by Twaticus — APPROVED** [review]

**Review by EOBGames — APPROVED** [review]
Well, the map stuff should be fine, but given the code changes, I'll leave this to an actual maintainer.

**Comment by YourNeighborsObnoxiousCat** [comment]
Nobody was using the mules for interaction-free deliveries anyway. Not when the parcel system (not the new chutes meta has) does its job a thousand times better. So honestly, yeah, get rid of them. A roundstart cargo only mech would make snagging crates from maint far less tedious and makes loading the shuttle a bit easier as well.

Though I will say that while it's cool to have the remaining mule tucked away somewhere, it would probably be best to not have it stuck behind a grille. If random intelligence rolls on the mule you'll have someone stuck in GBJ.

**Comment by Son-of-Space** [comment]
> 
> 
> Nobody was using the mules for interaction-free deliveries anyway. Not when the parcel system (not the new chutes meta has) does its job a thousand times better. So honestly, yeah, get rid of them. A roundstart cargo only mech would make snagging crates from maint far less tedious and makes loading the shuttle a bit easier as well.
> 
> Though I will say that while it's cool to have the remaining mule tucked away somewhere, it would probably be best to not have it stuck behind a grille. If random intelligence rolls on the mule you'll have someone stuck in GBJ.

already fixed

**Comment by cacogen** [comment]
Adding a mech is fine. The sprites are good as well.

I don't see the need to replace the MULEs though. They're fun to use, quick and easy.  If they need to tell the relevant departments they've delivered to them, that's something even I could add. 

But the MULEbot crate exists so if I want one on MetaStation I'll just order it.

**Comment by Son-of-Space** [comment]
> 
> 
> Adding a mech is fine. The sprites are good as well.
> 
> I don't see the need to replace the MULEs though. They're fun to use, quick and easy. If they need to tell the relevant departments they've delivered to them, that's something even I could add.
> 
> But the MULEbot crate exists so if I want one on MetaStation I'll just order it.

I welcome improvements to MULEbots, which will hopefully be a great compliment to this instead of a full replacement if they get a strong rework.

**Review by coiax — APPROVED** [review]
Code looks good to me.

**Comment by TheBonded** [comment]
> > Restrict the useable equipments on it so it can only use the hydraulic clamp, armour and power generators. no reason to give cargo a roundstart way of drilling open armoury crates
> > Also, maybe because it has a rusted frame, you can see through some damaged parts and see what it has loaded on inspection? just a thought
> > I think this is a step in the right direction for cargo
> 
> I'm nowhere near competent enough with byond code to implement what you're suggesting, but anybody else who comes around is welcome to make their own PR for it.

when this pr is merged I'll see about making a pr to restrict the available equipment. it seems equipment is told which mechs it can be equipped on in the equipment's code instead of the mech's code, which means I'll have to wait for this pr to be merged to have access to add '/obj/vehicle/sealed/mecha/working/ripley/cargo' to the exosuit drill code

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-5). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**
